File: backend/src/constants/hazard_severity.py
HazardSeverity = ["LOW", "MEDIUM", "HIGH", "CRITICAL"]
SEVERITY_RANK = {name: index for index, name in enumerate(HazardSeverity)}


def escalate(severity: str, target: str = "CRITICAL") -> str:
    """逾期升级：只升不降，至少升到 target。"""
    current = SEVERITY_RANK.get(severity, 0)
    floor = SEVERITY_RANK[target]
    return HazardSeverity[max(current, floor)]

File: frontend/src/constants/HazardSeverity.ts
export const HazardSeverity = ["LOW","MEDIUM","HIGH","CRITICAL"] as const;
export type HazardSeverity = (typeof HazardSeverity)[number];
export const HazardSeverityText: Record<HazardSeverity, string> = {
  LOW: "低",
  MEDIUM: "中",
  HIGH: "高",
  CRITICAL: "严重"
};
